### PR TITLE
Porting of 2D ControlNet tutorial using DDPM and ControlNet to guide image synthesis

### PR DESCRIPTION
**Pull request by virginiafdez**
Added the ControlNet (2D) notebook to the generation tutorials. The code uses BraTS, and conditions on binary brain masks. 

### Description
A few sentences describing the changes proposed in this pull request.

### Checks

- [X] Avoid including large-size files in the PR.
- [X] Clean up long text outputs from code cells in the notebook.
- [X] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [X ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [X] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

**Review by KumoLiu — APPROVED** [review]
Thanks for the PR, LGTM.
Please help update the README file here: https://github.com/Project-MONAI/tutorials/blob/main/generation/README.md? Thanks!

**Comment by KumoLiu** [comment]
Hi @virginiafdez, seems OOM error raised.
https://github.com/Project-MONAI/tutorials/actions/runs/10736878403/job/29777197959?pr=1815#step:5:840

File: generation/controlnet/2d_controlnet.ipynb [hunk 1]
```diff
@@ -0,0 +1,1190 @@
+{
```
Comment (ericspod) [inline generation/controlnet/2d_controlnet.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    train_loader = DataLoader(train_ds, batch_size=64, shuffle=True, num_workers=4, drop_last=True, persistent_workers=True)
</pre><p>For the CICD issues we can use a really small batch size like 16.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1815/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by KumoLiu** [comment]
Hi @virginiafdez, could you please help update the README under generation folder, then we can merge this one. Thanks!

**Comment by virginiafdez** [comment]
Done!!!!

Best wishes

Virginia
________________________________
